File: anstoss-transfer-market/src/main/java/de/mmagic/anstoss/anstosstransfermarket/model/Player.java
package de.mmagic.anstoss.anstosstransfermarket.model;

import de.mmagic.anstoss.anstosstransfermarket.store.PlayerStore;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.util.Map;

@Document(collection = PlayerStore.COLLECTION_NAME)
public class Player {

    public Player() {

    }

    public Player(Integer id, String name, Integer age, Double strength, String position, String country, Long price, Integer days, Map<String, List<Integer>> aaw) {
        this.id = id;
        this.name = name;
        this.age = age;
        this.strength = strength;
        this.position = position;
        this.country = country;
        this.price = price;
        this.days = days;
        this.aaw = aaw;
    }

    public Integer id;
    public String name;
    public Integer age;
    public Double strength;
    public String position;
    public String country;
    public Long price;
    public Integer days;
    public Map<String, List<Integer>> aaw;

}

File: anstoss-transfer-market-go/model/player.go
package model

type Player struct {
	Id       int     `bson:"_id" json:"id"`
	Age      int     `json:"age"`
	Name     string  `json:"name"`
	Position string  `json:"position"`
	Country  string  `json:"country"`
	Price    int     `json:"price"`
	Days     int     `json:"days"`
	Strength float64 `json:"strength"`
	Aaw      map[AawCategory][]int
}

type AawCategory string

const (
	Training AawCategory = "Training"
	Fitness  AawCategory = "Fitness"
	Einsatz  AawCategory = "Einsatz"
	Alter    AawCategory = "Alter"
	Tor      AawCategory = "Tor"
)
